src/components/lib/patina/patina_chan.c

```diff
@@ -0,0 +1,250 @@
+/**
+ * Redistribution of this file is permitted under the BSD two clause license.
+ *
+ * Copyright 2020, The George Washington University
+ * Author: Bite Ye, bitye@gwu.edu
+ */
+
+#include <llprint.h>
+#include <patina_chan.h>
+#include <patina_types.h>
+#include <static_slab.h>
+#include <string.h>
+
+/**
+ * This is a wrapper of patina's native channel library.
+ * For detail API usage guide, please check source codes
+ * under 'src/components/lib/chan'
+ */
+
+SS_STATIC_SLAB(chan, struct chan, PATINA_MAX_NUM_CHAN);
+SS_STATIC_SLAB(chan_rcv, struct chan_rcv, PATINA_MAX_NUM_CHAN);
+SS_STATIC_SLAB(chan_snd, struct chan_snd, PATINA_MAX_NUM_CHAN);
+
+/**
+ * Create a channel
+ *
+ * Currently we only support CHAN_DEFAULT as flag, hence
+ * the arugment 'flags' has no effect
+ * Also note that name as arugment is not supported.
+ *
+ * Arguments:
+ * - @type_size: size of the item in this channel
+ * - @queue_length: size of this channel
+ * - @ch_name: no effect
+ * - @falgs: no effect
+ *
+ * @return: id of the channel just been created.
+ */
+patina_chan_t
+patina_channel_create(size_t type_size, size_t queue_length, int ch_name, size_t flags)
+{
+	assert(type_size & queue_length);
+
+	struct chan *c = ss_chan_alloc();
+	assert(c);
+
+	assert(chan_init(c, type_size, queue_length, CHAN_DEFAULT));
+
+	ss_chan_activate(c);
+
+	return (patina_chan_t)c | PATINA_T_CHAN;
+}
```

Comment:
Note that both of the previous issues are repeated in some of the following functions/files. I won't post redundantly there so that you don't get drowned in notifications ;-)